### PR TITLE
BUG: IntervalIndex.get_indexer raising for read only array

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by WillAyd** [comment]
Should we just generally go back to using ndarray instead of memoryviews? If we set that expectation could be a good change for new contributors to pick up

**Comment by phofl** [comment]
This change is only necessary because cython does not support const with fused types unfortunately

**Comment by WillAyd** [comment]
Ah OK. So I guess the pattern is for fused types prefer ndarray (unless that gets fixed/implemented) elsewhere memview?

**Comment by phofl** [comment]
Yep that makes sense I guess

File: pandas/tests/indexes/interval/test_indexing.py [hunk 1]
```diff
@@ -424,6 +424,18 @@ def test_get_indexer_interval_index(self, box):
...
+        idx = interval_range(start=0, end=5)
+        arr = np.array([1, 2])
+        arr.flags.writeable = False
+        result = idx.get_indexer(arr)
+        expected = np.array([0, 1])
+        tm.assert_numpy_array_equal(result, expected)
+
+        result = idx.get_indexer_non_unique(arr)[0]
+        expected = np.array([0, 1])
+        tm.assert_numpy_array_equal(result, expected, check_dtype=False)
```
Comment (WillAyd) [inline pandas/tests/indexes/interval/test_indexing.py:437]: Hmm what's going on with the dtype? Wouldn't expect that issue

Comment (phofl) [reply pandas/tests/indexes/interval/test_indexing.py:437]: Windows/non windows stuff and since the dtype really doesnt matter here, I just ignored it

Comment (lithomas1) [reply pandas/tests/indexes/interval/test_indexing.py:437]: Maybe you can explicitly specify the type of the np.array call instead?

Comment (phofl) [reply pandas/tests/indexes/interval/test_indexing.py:436]: This would kill the 32 bit build... Open for suggestions but dtype really doesn't matter

**Comment by jorisvandenbossche** [comment]
>  So I guess the pattern is for fused types prefer ndarray (unless that gets fixed/implemented) elsewhere memview?

And that's also what we already do in most places. For example in groupby.pyx, you will see that we typically use ndarray for the input values (which use a fused type and can be const), while mostly use memoryviews for output values (which are writeable) or labels/mask (which don't use fused types). 
(although it's certainly not exact, I see some cases using ndarray that could use memoryview)

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by phofl** [comment]
merging this, happy to address comments in a follow up.

**Pull request merged**